packages/docregistry/src/default.ts

```diff
@@ -562,6 +562,8 @@ export class DocumentWidget<
     path: string
   ): void {
     this.title.label = PathExt.basename(sender.localPath);
+    // The document is not untitled any more.
+    this.isUntitled = false;
```

Comment:
Good idea. Though when user moves the new file to another directory (using file browser, without renaming it) it will also set `isUntitled` to `false`. Maybe the following would work?

```js
const newName = PathExt.basename(sender.localPath)
if (newName !== this.title.label) {
  this.isUntitled = false;
}
```

The current fix is sufficient for 3.5 release IMO (better for the new feature to not show up at edge cases than for it to show up when not needed annoying the user).

Reply:
Tracking in #13271.